lib/query/query_recurring_donations.rb

```diff
@@ -79,6 +84,13 @@ def self.full_search_expr(np_id, query)
       expr = expr.where("#{failed_or_active_clauses.join(' OR ')}")
     end
 
+    if include_last_failed_charge && query.key?(:started_at) && query.key?(:end_date)
```

Comment:
`:started_at` and `:end_date` are not the greatest hash names but I don't think you created them, did you?

Reply:
Yes! I created them. Those are to mark the beginning of the period and the ending of the period for failed recurring donations. I think I saw it like that somewhere else and reused it, but I think we can change that to something else, it would be better. What about "from_date" and "to_date"? Or "after_date" and "before_date"?

Reply:
So I'd like a way to indicate that the value for `:started_at` is inclusive of the time passed but `:end_date` is exclusive of the time passed. So what kind of names would we use to do that?

Reply:
Hmm, so how do `:from_date` and `:before_date` sound to you?

Reply:
I like it!